kusto/ingest/file_options.go

```diff
@@ -9,57 +9,58 @@ import (
 	"github.com/Azure/azure-kusto-go/kusto/ingest/internal/properties"
 )
 
-type OptionScope uint
+type TypeScope uint
+type ClientScope uint
 
 const (
-	IngestFromFile OptionScope = 1 << iota
-	IngestFromReader
-	IngestFromBlob
-	QueuedIngest
-	StreamingIngest
+	FromFile TypeScope = 1 << iota
+	FromReader
+	FromBlob
+	QueuedClient ClientScope = 1 << iota
+	StreamingClient
 )
 
-var scopesMap = map[OptionScope]string{
-	IngestFromFile:   "IngestFromFile",
-	IngestFromReader: "IngestFromReader",
-	IngestFromBlob:   "IngestFromBlob",
-	QueuedIngest:     "QueuedIngest",
-	StreamingIngest:  "StreamingIngest",
-}
-
 // FileOption is an optional argument to FromFile().
 type FileOption interface {
 	fmt.Stringer
 
-	OptionScopes() OptionScope
+	TypeScopes() TypeScope
+	ClientScopes() ClientScope
 
-	Run(p *properties.All, scopes OptionScope) error
+	Run(p *properties.All, clientType ClientScope, ingestionType TypeScope) error
 }
 
 type option struct {
-	run    func(p *properties.All) error
-	scopes OptionScope
-	name   string
+	run          func(p *properties.All) error
+	clientScopes ClientScope
+	typeScopes   TypeScope
+	name         string
+}
+
+func (o option) TypeScopes() TypeScope {
+	return o.typeScopes
+}
+
+func (o option) ClientScopes() ClientScope {
+	return o.clientScopes
 }
 
 func (o option) String() string {
 	return o.name
 }
 
-func (o option) OptionScopes() OptionScope {
-	return o.scopes
-}
+func (o option) Run(p *properties.All, clientType ClientScope, ingestionType TypeScope) error {
+	errType := errors.OpFileIngest
+	if clientType&StreamingClient != 0 {
+		errType = errors.OpIngestStream
+	}
 
-func (o option) Run(p *properties.All, scopes OptionScope) error {
+	if o.clientScopes&clientType == 0 {
+		return errors.ES(errType, errors.KClientArgs, fmt.Sprintf("%s is not valid for client type", o.name))
```

Comment:
Can you print the current client type and ingestion type (need some to string API)

Reply:
Done